packages/lib/src/Button.wc.svelte

```diff
@@ -26,33 +27,31 @@
 
   const variantClasses = {
     primary:
-      'bg-[rgb(var(--s-main-color))] text-white hover:brightness-90 focus:ring-[rgb(var(--s-main-color)/0.5)]',
+      'bg-[hsl(var(--s-main-color))] text-white hover:brightness-90 focus:ring-[hsla(var(--s-main-color),0.5)]',
     secondary:
-      'bg-transparent border border-[rgb(var(--s-main-color))] text-[rgb(var(--s-main-color))] hover:bg-[rgb(var(--s-main-color)/0.1)] focus:ring-[rgb(var(--s-main-color)/0.5)]',
+      'bg-transparent border border-[hsl(var(--s-main-color))] text-[hsl(var(--s-main-color))] hover:bg-[hsla(var(--s-main-color),0.1)] focus:ring-[hsl(var(--s-main-color))]/0.5',
     tertiary:
-      'text-[rgb(var(--s-main-color))] hover:brightness-75 focus:ring-[rgb(var(--s-main-color)/0.5)]'
+      'text-[hsl(var(--s-main-color))] hover:brightness-75 focus:ring-[hsla(var(--s-main-color),0.5)]'
   };
 
-  // Define reactive state for classes
-  let classes = $state('');
-
-  $effect(() => {
-    classes = twMerge(
+  let mergedClasses = $derived.by(() =>
+    twMerge(
       baseClasses,
       sizeClasses[size],
       variantClasses[variant],
-      disabled && 'opacity-50 cursor-not-allowed'
-    );
-  });
+      disabled && 'opacity-50 cursor-not-allowed',
+      classes
+    )
+  );
```

Comment:
@markusahlstrand compared to react stuff in `<script></script>`-tag doesn't every time the component gets reconciled.

If we would have:
```
<button class={twMerge(
      baseClasses,
      sizeClasses[size],
      variantClasses[variant],
      disabled && 'opacity-50 cursor-not-allowed',
      classes)}
/>
```

Each change to a prop would trigger reactivity. But since this was moved into it's own function Svelte wouldn't know values changed.

The way to deal with that is via `$derived` (previously known as `$:`).

It could be dealt with `$state` and `$effect` as you suggested, but that's going through unnessecary hoops since `$derived` meant for state that derives (state triggers state).